File: nodejs/checkio/src/home/second_index.ts
import assert from "assert";

function secondIndex(text: string, symbol: string): number | null {
	const firstIndex = text.indexOf(symbol);
	if (firstIndex === -1) return null;
	const secondIndex = text.indexOf(symbol, firstIndex + 1);
	return secondIndex === -1 ? null : secondIndex;
}

console.log("Example:");
console.log(secondIndex("sims", "s"));

// These "asserts" are used for self-checking
assert.strictEqual(secondIndex("sims", "s"), 3);
assert.strictEqual(secondIndex("find the river", "e"), 12);
assert.strictEqual(secondIndex("hi", " "), null);
assert.strictEqual(secondIndex("hi mayor", " "), null);
assert.strictEqual(secondIndex("hi mr Mayor", " "), 5);

console.log("Coding complete? Click 'Check Solution' to earn rewards!");

File: python/checkio/home/second-index.py
def second_index(text: str, symbol: str) -> int | None:
    try:
        return text.index(symbol, text.index(symbol) + 1)
    except ValueError:
        return None


print("Example:")
print(second_index("sims", "s"))

# These "asserts" are used for self-checking
assert second_index("sims", "s") == 3
assert second_index("find the river", "e") == 12
assert second_index("hi", " ") == None
assert second_index("hi mayor", " ") == None
assert second_index("hi mr Mayor", " ") == 5

print("The mission is done! Click 'Check Solution' to earn rewards!")
